drivers/regex/mlx5/host.c

```diff
@@ -1051,8 +1051,7 @@ int mlnx_poll(struct rxp_queue *rxp_queue, bool *rx_ready, bool *tx_ready)
         {
             //TODO: Check if can simply use work_id = "i", not sure if work_id
             //      maps SQ's if only 1 job per SQ!?
-            ret = mlx5_regex_poll(rxp_queue->rxp_job_ctx, i,
-                                   rxp_queue->sq_buf[i].work_id);
+            mlx5_regex_poll(rxp_queue->rxp_job_ctx, i);
```

Comment:
shouldn't you take in to account the returned index from mlx5_regex_poll?

Reply:
yes, this is a bug, I will fix